### PR TITLE
NMS-13123: Prevent HTML markup in notification wizard

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-13123

### CODE REVIEW

**Comment by christianpape** [comment]
I escaped the strings for the forms and buttons. The question is: should I still prevent the user from submitting the form with HTML/JS characters in the name?

File: opennms-webapp/src/main/webapp/WEB-INF/jsp/admin/notification/noticeWizard/eventNotices.jsp [hunk 1]
```diff
@@ -109,31 +109,31 @@
           <c:forEach items="${notifications}" var="notification">
           <tr>
             <td>
-              <input type="button" class="btn btn-secondary" value="Edit" onclick="javascript:editNotice('${notification.escapedName}')"/>
+              <input type="button" class="btn btn-secondary" value="Edit" onclick="javascript:editNotice('${fn:escapeXml(notification.escapedName)}')"/>
```
Comment (j-white) [inline opennms-webapp/src/main/webapp/WEB-INF/jsp/admin/notification/noticeWizard/eventNotices.jsp:112]: Escaping for JavaScript vs XML would be safer here. See https://github.com/OpenNMS/opennms/pull/1763/files#diff-22f9d3c942102d20d541afa9a9aa46ea3a31f641449162efa61ad774684f33e4R101

**Review by j-white — COMMENTED** [review]
I don't think preventing submissions with special characters is necessary if everything is escaped property on render.

If we do enforce validation rules, then other endpoints like REST APIs should enforce those as well.

**Comment by j-white** [comment]
Patch looks good. Should we backport this to `foundation-2020` so that there is a Meridian release with the patch?

**Pull request merged**